cmd/juju/application/config_test.go

```diff
@@ -169,7 +169,19 @@ func (s *configCommandSuite) TestSetCommandInit(c *gc.C) {
 
 	// --reset and no config name provided
 	err = coretesting.InitCommand(application.NewConfigCommandForTest(s.fake), []string{"application", "--reset"})
-	c.Assert(err, gc.ErrorMatches, "no configuration options specified")
+	c.Assert(err, gc.ErrorMatches, "flag needs an argument: --reset")
```

Comment:
I might be mistaken but I believe the "can only retrieve one or all... " path is not tested.